File: 4242-sum-of-gcd-of-formed-pairs/sum-of-gcd-of-formed-pairs.py
class Solution:
    def gcdSum(self, nums: list[int]) -> int:
        prefixgcd=[]
        mx=-1
        for i in nums :
            mx=max(mx,i)
            prefixgcd.append(gcd(mx,i))
        prefixgcd.sort()
        i=0
        j=len(nums)-1
        ans=0
        while i<j:
            ans+=gcd(prefixgcd[i],prefixgcd[j])
            i+=1
            j-=1
        return ans    

File: 4242-sum-of-gcd-of-formed-pairs/sum-of-gcd-of-formed-pairs.cpp
class Solution {
public:
    long long gcdSum(vector<int>& nums) {
        int i, j, n = nums.size();
        vector<int> prefixgcd(n);
        int mx = -1;
        for (i = 0; i < n; i++) {
            mx = max(nums[i], mx);
            prefixgcd[i] = __gcd(mx, nums[i]);
        }
        sort(prefixgcd.begin(), prefixgcd.end());
        i = 0;
        j = n - 1;
        long long ans = 0;
        while (i < j) {
            ans += __gcd(prefixgcd[i], prefixgcd[j]);
            i++;
            j--;
        }
        return ans;
    }
};